blocks/model.py

```diff
@@ -190,7 +179,13 @@ def get_params(self):
         parameters that do not belong to any brick.
 
         """
-        return self.params
+        if (len(set(parameter.name for parameter in self.parameters)) !=
+                len(self.parameters)):
+            raise ValueError("parameters should have different names")
```

Comment:
Thanks, I see now what the problem is. If you read the previous version of the code, you will use that hierarchical names of the parameters were used. You can get those by using `Brick.get_unique_path`.